### DISCUSSION TITLE
Using the classification models

### DISCUSSION THREAD

**Discussion by elvijsm2** [mindee/doctr#1128] [Q&A]
Hi! I have a bunch of images of individual character crops which I would like to classify. I see that there are a bunch of pre-trained classification models with checkpoints in the classification model zoo. Unfortunately, I have not been able to figure out how to use them.

In particular, regardless of which classification model from ```doctr.models.classification``` I choose, I get results no better than from a randomly initialized model:
```
from doctr.models.classification import mobilenet_v3_small
predictor = mobilenet_v3_small(pretrained=True, include_top=True)

# Downloads https://doctr-static.mindee.com/models?id=v0.4.1/mobilenet_v3_small-8a32c32c.zip&src=0 to ~/.cache/doctr/models/mobilenet_v3_small-8a32c32c.zip
```
Running the predictor on any (32, 32, 3) image as follows
```
out = predictor(image[np.newaxis])
```
produces a (1, 126) tensor output, which makes sense given the French vocabulary of 126 characters. But the values are seemingly random and have nothing to do with the image content. I've tried on my own images, as well as clean images produced by the ```CharacterGenerator``` from ```doctr.datasets```. I also tried preprocessing/normalizing the images, but never got anything sensible. 

What does work is using a pre-trained ```recognition_predictor```. But what it ends up doing is putting the 32x32 patch on a 32x128 canvas (which otherwise consists of all zeros), and postprocessing the model output with CTC. The "classification" output obtained this way is pretty good. But this is obviously very wasteful.

I'm using v0.6. I must be missing something obvious and would appreciate some guidance.

**Comment by elvijsm2** [comment]
Shamelessly bumping this. I would really appreciate some tips. Thanks!

**Reply by felixdittrich92** [reply]
Hi @elvijsm2 👋 

To get the models prediction you need to run:

`pred = torch.argmax(out, dim=-1)`

At the end all classification models are only trained on synthetic data to provide a pretrained backbone for the recognition/detection architectures on top, so it was not the goal to provide this models as standalone. :) 
For a "real" case classification task you should finetune your prefered classification model (We provide a script in the references folder https://github.com/mindee/doctr/tree/main/references/classification )
